syntax = "proto3";

package order_queue;

service OrderQueue {
  rpc EnqueueOrder (EnqueueRequest) returns (EnqueueResponse);
  rpc DequeueOrder (DequeueRequest) returns (DequeueResponse);
}

message EnqueueRequest {
  string orderId = 1;
}

message EnqueueResponse {
  string orderId = 1;
  bool Enqueued = 2;
}

message DequeueRequest {}

message DequeueResponse {
  string orderId = 1;
  bool Dequeued = 2;
}
